bundler/index.js

```diff
@@ -4,70 +4,115 @@ const HtmlParser = require('./modules/htmlParser');
 
 class Bundler {
   constructor(
-    fileManager = new FileManager(),
-    uglifier = new Uglifier(),
-    htmlParser = new HtmlParser()
+    pathConfig,
+    tagConfig,
   ) {
-    this.fileManager = fileManager;
-    this.uglifier = uglifier;
-    this.htmlParser = htmlParser;
+    this.pathConfig = pathConfig;
+    this.tagConfig = tagConfig;
+    this.fileManager = new FileManager();
+    this.uglifier = new Uglifier();
+    this.htmlParser = new HtmlParser(tagConfig);
   }
 
-  async createBundle(htmlFilePath, schema = null) {
-    if (!schema) {
-      schema = this.createSchema(htmlFilePath);
+  async create() {
+    try {
+      this._logBundleStarting();
+      this._dist();
+      this._assets();
+      const htmlPath = this.pathConfig.HTML;
+      const htmlData = await this.fileManager.readAggregateFilesData(htmlPath);
+      if (!htmlData) {
+        this._logFileNotCreated(htmlPath);
+        throw `File not created ${htmlPath}`;
+      }
+      const cssBundleName = await this._css(htmlData);
+      const jsBundleName = await this._js(htmlData);
+      await this._html(htmlData, htmlPath, cssBundleName, jsBundleName);
+      this._logBundleFinished();
+    } catch(error) {
+      this._removeDistFolder();
+      this._logBundleFailed();
     }
-    // html
-    await this.processBundleUnit(
-      schema.HTML.input,
-      schema.HTML.output,
-      this.uglifier.uglifyHtml,
-      true
-    );
-    // css
-    await this.processBundleUnit(
-      schema.CSS.input,
-      schema.CSS.output,
-      this.uglifier.uglifyCss
-    );
-    // js
-    await this.processBundleUnit(
-      schema.JS.input,
-      schema.JS.output,
-      this.uglifier.uglifyJs
-    );
-  }
-
-  createSchema(htmlFilePath) {
-    const rawPathes = this.htmlParser.findPathesInHtml(htmlFilePath);
-    
-    const pathes = {};
-    Object.keys(rawPathes).forEach((key) => {
-      pathes[key] = {
-        input: rawPathes[key],
-        output: `dist/bundle.${key.toLowerCase()}`
-      };
-    });
-
-    const split = htmlFilePath.split('/');
-    const htmlFileName = split[split.length - 1];
-    pathes.HTML = {
-      input: [ htmlFilePath ],
-      output: `dist/${htmlFileName}`
-    };
+  }
+
+  _logBundleStarting() {
```

Comment:
It would be great to have all these private `log` methods in a separate module. For example "Logger".

This way we will clean the `Bundler` class from the technical implementation of our logs, and delegate it to another piece of the system.